### PR TITLE
8315882: Open some swing tests 2

### PR DESCRIPTION
**Pull request by alisenchung**
Moving some tests to open
5 javax/swing/ScrollPaneLayout/4688907/bug4688907.java
6 javax/swing/SpringLayout/4756178/bug4756178.java
7 javax/swing/SpringLayout/4803649/bug4803649.java

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8315882](https://bugs.openjdk.org/browse/JDK-8315882): Open some swing tests 2 (**Bug** - P4)

### Reviewers
 * [Harshitha Onkar](https://openjdk.org/census#honkar) (@honkar-jdk - Committer) ⚠️ Review applies to [ae558c94](https://git.openjdk.org/jdk/pull/15850/files/ae558c94c8d64d397673174bcfbd40d390d9e42b)
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**) ⚠️ Review applies to [ae558c94](https://git.openjdk.org/jdk/pull/15850/files/ae558c94c8d64d397673174bcfbd40d390d9e42b)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15850/head:pull/15850` \
`$ git checkout pull/15850`

Update a local copy of the PR: \
`$ git checkout pull/15850` \
`$ git pull https://git.openjdk.org/jdk.git pull/15850/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15850`

View PR using the GUI difftool: \
`$ git pr show -t 15850`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15850.diff">https://git.openjdk.org/jdk/pull/15850.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15850#issuecomment-1728506274)

### CODE REVIEW

File: test/jdk/javax/swing/JScrollPane/bug4688907.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ */
+
+/* @test
+ * @bug 4688907
+ * @summary ScrollPaneLayout.minimumLayoutSize incorrectly compares hsbPolicy
+ */
+
+import java.awt.Dimension;
+import javax.swing.JScrollPane;
+import javax.swing.SwingUtilities;
```
Comment (honkar-jdk) [inline test/jdk/javax/swing/JScrollPane/bug4688907.java:31]: Remove unused import

File: test/jdk/javax/swing/SpringLayout/bug4756178.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                bt = new JButton();
+
+                buttonPreferredSize = new Dimension(20, 20);
+                bt.setPreferredSize(buttonPreferredSize);
+                p.add(bt, cc);
+
+                fr.setSize(200, 200);
+                fr.setLocationRelativeTo(null);
+                fr.setVisible(true);
+
```
Comment (honkar-jdk) [inline test/jdk/javax/swing/SpringLayout/bug4756178.java:68]: Extra new line can be removed.

File: test/jdk/javax/swing/SpringLayout/bug4803649.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JTextArea;
+import javax.swing.Spring;
+import javax.swing.SpringLayout;
+import javax.swing.SwingUtilities;
+
+public class bug4803649 {
+    static JFrame fr;
+    static JPanel panel;
```
Comment (honkar-jdk) [inline test/jdk/javax/swing/SpringLayout/bug4803649.java:41]: panel can be made local var.

File: test/jdk/javax/swing/SpringLayout/bug4756178.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            });
+
+            if (!buttonPreferredSize.equals(bt.getSize())) {
+                throw new RuntimeException("Button size is " + bt.getSize() +
+                        ", should be " + buttonPreferredSize);
+            }
```
Comment (honkar-jdk) [inline test/jdk/javax/swing/SpringLayout/bug4756178.java:74]: Delay (`Thread.sleep()`) should be add after frame is shown and bt.getSize() should be wrapped in EDT.
The delay ensures the frame is updated before querying the button size.

```suggestion
          Thread.sleep(1000);
            SwingUtilities.invokeAndWait(() -> actualSize = bt.getSize());
            if (!buttonPreferredSize.equals(actualSize)) {
                throw new RuntimeException("Button size is " + actualSize +
                        ", should be " + buttonPreferredSize);
            }
```

**Review by honkar-jdk — CHANGES_REQUESTED** [review]

File: test/jdk/javax/swing/SpringLayout/bug4803649.java (hunk 3 above)
Comment (alisenchung) [reply test/jdk/javax/swing/SpringLayout/bug4803649.java:41]: looks like panel is used later to check the dimension, so it has to be a global var

**Review by honkar-jdk — APPROVED** [review]
Updated changes look good.

**Review by prrace — APPROVED** [review]

**Pull request closed without merging**